From 32224f62589080a2098334262d95f4c476eebc1b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:20 -0400
Subject: [usb] xhci: use completion event's slot id rather than dig it out of command

Message-id: <1410887463-20674-69-git-send-email-dzickus@redhat.com>
Patchwork-id: 95140
O-Subject: [RHEL7 PATCH 068/591] xhci: use completion event's slot id rather than dig it out of command
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit 20e7acb13ff48fbc884d5918c3697c27de63922a
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 13:29:50 2013 +0300

    xhci: use completion event's slot id rather than dig it out of command

    Since the slot id retrieved from the Reset Device TRB matches the slot id in
    the command completion event, which is available, there is no need to determine
    it again.
    This patch removes the uneccessary reassignment to slot id and adds a WARN_ON
    in case the two Slot ID fields differ (although according xhci spec rev1.0
    they should not differ).

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index b85fbcb..1c3b3f4 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1547,9 +1547,9 @@ bandwidth_change:
   xhci_handle_cmd_reset_ep(xhci, event, xhci->cmd_ring->dequeue);
   break;
  case TRB_TYPE(TRB_RESET_DEV):
+  WARN_ON(slot_id != TRB_TO_SLOT_ID(
+    le32_to_cpu(xhci->cmd_ring->dequeue->generic.field[3])));
   xhci_dbg(xhci, "Completed reset device command.\n");
-  slot_id = TRB_TO_SLOT_ID(
-   le32_to_cpu(xhci->cmd_ring->dequeue->generic.field[3]));
   virt_dev = xhci->devs[slot_id];
   if (virt_dev)
    handle_cmd_in_cmd_wait_list(xhci, virt_dev, event);
-- 
1.7.1